File: SOURCE/EFEX/BASE/MICROSOFT/C#/cMessage.cs
/// <summary>
/// Type   : Class
/// Name   : cMessage
/// Author : Softstep Pty Ltd
/// Date   : July 2008
/// </summary>
namespace EfexServer {
	
   using System;
	
	/// <summary>
	/// This class defines a message
	/// </summary>
	public class cMessage {

		//
		// Class declarations
		//
		private char cchrCode;
		private string cstrData;
		
		/// <summary>
		/// Constructs a new instance
		/// </summary>
		internal cMessage(char chrCode, string strData) {
			cchrCode = chrCode;
			cstrData = strData;
			if (cstrData == null) {
				cstrData = "";
			}
		}
		
		/// <summary>
		/// Gets the mailbox message code
		/// </summary>
		/// <returns>char the mailbox message code</returns>
      internal char GetCode() {
         return cchrCode;
      }

		/// <summary>
		/// Gets the mailbox message length
      /// </summary>
		/// <returns>short the mailbox message length</returns>
      internal short GetLength() {
         return (short)cstrData.Length;
      }

		/// <summary>
		/// Gets the mailbox message data
		/// </summary>
		/// <returns>string the mailbox message data</returns>
      internal string GetData() {
         return cstrData;
      }

		/// <summary>
		/// Gets the mailbox message - code, length and data
		/// </summary>
		/// <returns>string the mailbox message data</returns>
      internal string GetMessageData() {
         return new string((cchrCode.ToString() + ((char)cstrData.Length).ToString() + cstrData).ToCharArray());
      }

   }

}

File: SOURCE/EFEX/BASE/JAVA/cMessage.java
/**
 * Package : ISI Efex
 * Type    : Class
 * Name    : cMessage
 * Author  : Steve Gregan
 * Date    : June 2008
 */
package com.isi.efex;

/**
 * This class defines a message
 */
public class cMessage {

   //
   // Class declarations
   //
   private char cchrCode;
   private String cobjData;
   
   /**
    * Constructs a new instance
    */
   public cMessage(char chrCode, String objData) {
      cchrCode = chrCode;
      cobjData = objData;
      if (cobjData == null) {
         cobjData = "";
      }
   }
   
   /**
    * Gets the mailbox message code
    * 
    * @return char the mailbox message code
    */
   public char getCode() {
      return cchrCode;
   }
   
   /**
    * Gets the mailbox message length
    * 
    * @return short the mailbox message length
    */
   public short getLength() {
      return (short)cobjData.length();
   }
   
   /**
    * Gets the mailbox message data
    * 
    * @return String the mailbox message data
    */
   public String getData() {
      return cobjData;
   }
   
   /**
    * Gets the mailbox message - code, length and data
    * 
    * @return String the mailbox message data
    */
   public String getMessageData() {
      return new String(String.valueOf(cchrCode) + String.valueOf((char)(short)cobjData.length()) + cobjData);
   }

}